### 1번. 짝수를 찾아라 [3058]

In [9]:
t = int(input())

num = []
for _ in range(t):
    num = list(map(int,input().split()))
    
    results = []
    for i in num:
        if i%2 == 0:
            results.append(i)
    results.sort()
    
    print(sum(results),results[0])

2
1 2 3 4 5 6 7
12 2
13 78 39 42 54 93 86
260 42


### 2번. 루트 [4619]

In [19]:
import math
while True:
    b,n =  map(int,input().split())
    if b == 0 and n == 0:
        break
    a = math.floor(b**(1/n))
    a1 = abs(b - (a-1)**n) 
    a2 = abs(b - a**n)
    a3 = abs(b- (a+1)**n)
    a_find = min(a1,a2,a3)
    if a_find == a1:
        print(a-1)
    elif a_find == a2:
        print(a)
    else:
        print(a+1)

4 3
1
0 0


### 3번. 다음수 [4880]

In [26]:
while True:
    a1,a2,a3 = map(int, input().split())
    if a1 == 0 and a2 == 0 and a3 == 0:
        break
    a_ap = [a2-a1, a3-a2]
    if len(set(a_ap)) == 1:
        print("AP",a3+(a3-a2))
    else: 
        if a1 !=0 and a2 !=0:
            a_gp = [a2//a1, a3//a2]
        if len(set(a_gp)) == 1:
            print("GP",a3*(a3//a2))

-3 0 3
AP 6
-9 -3 0
GP 0
0 0 0


### 4번. 완전 세제곱 [4690]

In [44]:
for a_ in range(2,101):
    for b in range(2,101):
        for c in range(b+1,101):
            for d in range(c+1,101):
                if (a_**3) == (b**3) + (c**3) + (d**3):
                        print("Cube = %s, Triple = (%s,%s,%s)" % (a_,b,c,d))

Cube = 6, Triple = (3,4,5)
Cube = 12, Triple = (6,8,10)
Cube = 18, Triple = (2,12,16)
Cube = 18, Triple = (9,12,15)
Cube = 19, Triple = (3,10,18)
Cube = 20, Triple = (7,14,17)
Cube = 24, Triple = (12,16,20)
Cube = 25, Triple = (4,17,22)
Cube = 27, Triple = (3,18,24)
Cube = 28, Triple = (18,19,21)
Cube = 29, Triple = (11,15,27)
Cube = 30, Triple = (15,20,25)
Cube = 36, Triple = (4,24,32)
Cube = 36, Triple = (18,24,30)
Cube = 38, Triple = (6,20,36)
Cube = 40, Triple = (14,28,34)
Cube = 41, Triple = (2,17,40)
Cube = 41, Triple = (6,32,33)
Cube = 42, Triple = (21,28,35)
Cube = 44, Triple = (16,23,41)
Cube = 45, Triple = (5,30,40)
Cube = 46, Triple = (3,36,37)
Cube = 46, Triple = (27,30,37)
Cube = 48, Triple = (24,32,40)
Cube = 50, Triple = (8,34,44)
Cube = 53, Triple = (29,34,44)
Cube = 54, Triple = (6,36,48)
Cube = 54, Triple = (12,19,53)
Cube = 54, Triple = (27,36,45)
Cube = 56, Triple = (36,38,42)
Cube = 57, Triple = (9,30,54)
Cube = 58, Triple = (15,42,49)
Cube = 58, Triple = (22,30,54

### 5번. 얼마? [9325]

In [47]:
t = int(input())


for _ in range(t):
    s = int(input())
    n = int(input())
    option = []
    for i in range(n):
        q,p = map(int, input().split())
        option.append(q*p)
        
    print(s+sum(option))

2
10000
2
1 2000
3 400
13200
50000
0
50000


### 6번. 더하기 [10822]

In [48]:
a = list(map(int, input().split(",")))
print(sum(a))

10,20,30,50,100
210


### 7번. NN [11944]

In [53]:
n, m = map(int, input().split())
print((str(n)*n)[:m])

20 16
2020202020202020


### 8번. 닉네임에 갓 붙이기 [13163]

In [56]:
n = int(input())

for _ in range(n):
    name = list(input().split())
    name[0] = 'god'
    print(''.join(name))

1
beak joon
godjoon


### 9번. 거꾸로 구구단 [13410]

In [60]:
n,k = map(int, input().split())

gugudan = []
for j in range(1,k+1,1):
    gugudan.append(n*j)
max_gugudan = []
for i in gugudan:
    max_gugudan.append(int(str(i)[::-1]))
    
print(max(max_gugudan))

8 9
84


### 10번. 차량 번호판 1 [16968]

In [118]:
car = input()
dic = {'d':10, 'c':26}
all1 = 10
all2 = 26
temp2 = 1

for i in range(0,len(car),1):
    if i == 0:
        temp2 *= dic[car[i]]
    if i >= 1:
        if car[i] == car[i-1]:
            if car[i] == 'd':
                if all1 == 10:
                    all1 -= 1
                    temp2 *= (all1)
                else :
                    temp2 *= 9
            if car[i] == 'c':
                if all2 == 26:
                    all2 -= 1
                    temp2 *= (all2)
                else :
                    temp2 *= 25
        
        else:
            temp2 *= dic[car[i]]
print(temp2)

dddd
7290


연속으로 오는 두 자리만 신경써야 함. 따라서 dddd 연속으로 오면 10*9*9*9 이렇게 곱해져서 7290이 된다.

### 11번. 막대기 [1094]

In [115]:
x = int(input())

stick = [64,32,16,8,4,2,1]
cnt = 0

for i in range(len(stick)):
    if x >= stick[i]:
        cnt += 1
        x -= stick[i]
        
    if x == 0:
        break
        
print(cnt)

23
16
4
2
1
4


### 12번. 영화감독 숌 [1436]

문제 이해가 중요. 666 숫자가 나올 때마다 카운터를 늘리면 된다.

In [67]:
num = int(input())
n = 0
for i in range(665,10000000):
    if '666' in str(i):
        n += 1   
    if num == n :
        print(i)
        break


500
166699


-> 질문. 여기 범위를 다르게 해주거나 while문 쓴 코드 궁금

### 13번. 방 번호 [1475]

참고 : https://bgspro.tistory.com/62

In [141]:
import math
n = list(map(str,input()))
one_set = [0]*10
for i in n:
    if i == '9' or i == '6':
        one_set[6] += 0.5
    else:
        one_set[int(i)] += 1

one_set[6] = math.ceil(one_set[6])


print(max(one_set))

9966
2


-> 이거 왜 안되는지 질문하기

In [69]:
word = input()
ans = [0] * 10
for i in range(len(word)):
    num = int(word[i])
    if num == 6 or num == 9:
        if ans[6] <= ans[9]:
            ans[6] += 1
        else:
            ans[9] += 1
    else:
        ans[num] += 1

print(max(ans))

122
2


### 14번. 소수 찾기 [1978]

In [115]:
n = int(input())
a = list(map(int, input().split()))

sosu = []
for i in range(2,1001):
    term = []
    for j in range(i,0,-1):
        if i%j == 0:
            term.append(j)
            if len(term) == 3:
                pass
    if len(term) == 2:
        sosu.append(i)
    
    
    
count = 0
for a_ in a:
    if a_ in sosu:
        count += 1
        
print(count)

4
1 3 5 7
3


In [119]:
num = []
for i in sosu:
    if 60<=i<=100:
        num.append(i)
result = [min(num),sum(num)]
for i in result:
    print(i)

False

### 15번. 소수 [2581]

In [125]:
m = int(input())
n = int(input())

sosu = []
for i in range(2,10001):
    term = []
    for j in range(i,0,-1):
        if i%j == 0:
            term.append(j)
            if len(term) >= 3:
                pass
    if len(term) == 2:
        sosu.append(i)
        
num = []
for i in sosu:
    if m<=i<=n:
        num.append(i)
if len(num) == 0:
    print(-1)
else :
    result = [sum(num),min(num)]
    for i in result:
        print(i)


64
65
-1


### 16번. 에라토스테네스의 체 [2960]

소수를 찾는 방법. 범위에서 합성수를 지우는 방식으로 소수를 찾는다. 

In [1]:
n=30
a = [False,False] + [True]*(n-1)
primes=[]

for i in range(2,n+1):
    if a[i]:
        primes.append(i)
    for j in range(2*i, n+1, i):
        a[j] = False
print(primes)

[2, 3, 5, 7, 11, 13, 17, 19, 23, 29]


https://velog.io/@jajubal/%ED%8C%8C%EC%9D%B4%EC%8D%AC%EB%B0%B1%EC%A4%80-2960-%EC%97%90%EB%9D%BC%ED%86%A0%EC%8A%A4%ED%85%8C%EB%84%A4%EC%8A%A4%EC%9D%98-%EC%B2%B4

In [133]:
n,k = map(int,input().split())
temp = 0
dis = [True]*(n+1)

for i in range(2,n+2):
    for j in range(i, n+1, i):
        if dis[j] != False:
            dis[j] = False
            temp += 1
            if temp == k:
                print(j)
                break

10 7
9


In [ ]:
N, K = map(int, input().split())

cnt = 0

nums = [True] * (N + 1)
for i in range(2, len(nums) + 1):
    for j in range(i, N+1, i):
        if nums[j] == True:
            nums[j] = False
            cnt = cnt + 1
            if cnt == K:
                print(j)
                break

틀린거 찾아줘 ,,,

### 17번. 덩치 [7568]

자기 자신보다 둘 다 작으면 count 해주기

In [120]:
n = int(input())

dungchi = []
for _ in range(n) :
    dungchi.append(list(map(int,input().split())))

for i in dungchi:
    rank = 1
    for j in dungchi:
        if i[0] < j[0] and i[1]<j[1]:
            rank += 1
    print(rank, end=" ")

5
55 185
58 183
88 186
60 175
46 155
2 2 1 2 5 

### 18번. 올림픽 [8979]

In [9]:
n,k = map(int,input().split())

temp = []
ct =[]
for _ in range(n):
    country = list(map(int,input().split()))
    ct.append(country[0])


for j in range(len(ct)):
    if ct[j][1]

4 3
1 1 2 0
2 0 1 0
3 0 1 0
4 0 0 1


In [123]:
N,K=map(int,input().split())
l=[]
for i in range(N):
    l.append(list(map(int,input().split())))
L=sorted(l, key=lambda x: (-x[1],-x[2],-x[3])) # 순서대로 정렬됨
for i in range(N):
    if L[i][0]==K:
        rank=i
for j in range(N):
    if L[rank][1:]==L[j][1:]:
        print(j+1)
        break

4 3
1 1 2 0
2 0 1 0
3 0 1 0
4 0 0 1
2


### 19번. 오늘도 졌다 [14582]

In [25]:
num = []
for _ in range(2):
    a = list(map(int, input().split()))
    num.append(a)
     
num_a = 0
num_b = 0
results = []
for i in range(9):
    num_a += num[0][i]
    if num_a > num_b:
        results.append("Yes")
    num_b += num[1][i]

if num_a < num_b and "Yes" in set(results) : 
    print("Yes")
else:
    print("No")

1 0 0 0 0 0 2 2 1
0 0 3 0 0 0 0 1 4
YES


-> 왜 틀렸는지 모르겠음! -> 대소문자 ..

In [29]:
j=list(map(int,input().split()))
s=list(map(int, input().split()))
jsum=0
ssum=0
chk=False
for i in range(9):
    jsum+=j[i]
    if jsum>ssum:
        chk=True
    ssum+=s[i]
if jsum<ssum and chk==True:
    print('Yes')
else:
    print('No')

1 0 0 0 0 0 2 2 1
0 0 3 0 0 0 0 1 4
Yes


### 20번. 세 개의 소수 문제 [11502]

In [ ]:
t = int(input())

def isPrime(a):
    if(a<2):
        return False
    for i in range(2,a):
        if(a%i==0):
            return False
        return True
    
for _ in range(t):
    a = int(input())
    a_sosu = [2,]
    for i in range(3,a+1,1):
        if isPrime(i)== True:
            a_sosu.append(i)

In [157]:
t = int(input())
for _ in range(t):
    n = int(input())
    a = [False,False] + [True]*(n)
    a_sosu=[]

    for i in range(2,n+1):
        if a[i]:
            a_sosu.append(i)
        for j in range(2*i, n+1, i):
            a[j] = False
    
    num = []
    for a_ in a_sosu:
        for b_ in a_sosu:
            for c_ in a_sosu:
                if n == a_+b_+c_:
                    num.append([a_,b_,c_])
                
                if len(num) == 1:
                    break
            if len(num) == 1:
                    break
        if len(num) == 1:
                    break
                
    if not num:
        print(0)
    else:
        print(num[0][0],num[0][1],num[0][2],sep =" ")              

3
7
2 2 3
11
2 2 7
25
3 3 19


-> 시간초과 ㅠ.ㅠ

In [2]:
N=int(input())
l=[]
for i in range(N):
    l.append(int(input()))
ll=[]
c=2
while c != max(l):
    lll=[]
    for i in range(len(ll)):
        lll.append(c%ll[i])
    if 0 not in lll :
        ll.append(c)
    c+=1
for s in l:    
    for i in ll:
        for ii in ll:
            for iii in ll:
                if s==i+ii+iii:
                    ans=[i,ii,iii]
                    break
                else:
                    ans=[0]
            if len(ans)==3:
                break
        if len(ans)==3:
            break
    print(*ans)